lib-ee/emqx_ee_schema_registry/src/emqx_ee_schema_registry_http_api.erl

```diff
@@ -0,0 +1,251 @@
+        put => #{
+            tags => [<<"schema_registry">>],
+            summary => <<"Update a schema">>,
+            description => ?DESC("desc_schema_registry_api_put"),
+            parameters => [param_path_schema_name()],
+            'requestBody' => emqx_dashboard_swagger:schema_with_examples(
+                emqx_ee_schema_registry_schema:api_schema("put"),
+                post_examples()
+            ),
+            responses =>
+                #{
+                    200 =>
+                        emqx_dashboard_swagger:schema_with_examples(
+                            emqx_ee_schema_registry_schema:api_schema("put"),
+                            put_examples()
+                        ),
+                    404 => error_schema('NOT_FOUND', "Schema not found")
+                }
+        },
+        delete => #{
+            tags => [<<"schema_registry">>],
+            summary => <<"Delete registered schema">>,
+            description => ?DESC("desc_schema_registry_api_delete"),
+            parameters => [param_path_schema_name()],
+            responses =>
+                #{
+                    204 => <<"Schema deleted">>,
+                    404 => error_schema('NOT_FOUND', "Schema not found")
+                }
+        }
+    }.
+
+%%-------------------------------------------------------------------------------------------------
+%% API
+%%-------------------------------------------------------------------------------------------------
+
+'/schema_registry'(get, _Params) ->
+    Schemas = emqx_ee_schema_registry:list_schemas(),
+    Response =
+        lists:map(
+            fun({Name, Params}) ->
+                Params#{name => Name}
+            end,
+            maps:to_list(Schemas)
+        ),
+    ?OK(Response);
+'/schema_registry'(post, #{body := Params0 = #{<<"name">> := Name}}) ->
+    Params = maps:without([<<"name">>], Params0),
+    case emqx_config:get([?CONF_KEY_ROOT, schemas, Name], undefined) of
+        undefined ->
+            case emqx_ee_schema_registry:add_schema(Name, Params) of
+                ok ->
+                    Res = emqx_config:get([?CONF_KEY_ROOT, schemas, Name]),
+                    {201, Res#{name => Name}};
+                {error, Error} ->
+                    ?BAD_REQUEST(Error)
+            end;
+        _ ->
+            ?BAD_REQUEST('ALREADY_EXISTS', <<"Schema already exists">>)
+    end.
+
+'/schema_registry/:name'(get, #{bindings := #{name := Name}}) ->
+    case emqx_config:get([?CONF_KEY_ROOT, schemas, Name], undefined) of
+        undefined ->
+            ?NOT_FOUND(<<"Schema not found">>);
+        Res ->
+            ?OK(Res#{name => Name})
+    end;
+'/schema_registry/:name'(put, #{bindings := #{name := Name}, body := Params}) ->
+    case emqx_config:get([?CONF_KEY_ROOT, schemas, Name], undefined) of
+        undefined ->
+            ?NOT_FOUND(<<"Schema not found">>);
+        _ ->
+            case emqx_ee_schema_registry:add_schema(Name, Params) of
```

Comment:
Nit: we assume `emqx_ee_schema_registry:add_schema/2` implicit effects here (config changes).

Mb. use emqx_ee_schema_registry API for accessing the schemas too?
I.e. smth. like
```
maps:get(Name, emqx_ee_schema_registry:list_schemas(), undefined)
```
instead of
```
emqx_config:get([?CONF_KEY_ROOT, schemas, Name], undefined)
```
?

Reply:
:heavy_check_mark: